src/components/ChatRoomListItem.js

```diff
@@ -0,0 +1,89 @@
+import React from 'react';
+import styled from 'styled-components';
+
+const StyledChatRoomListItem = styled.div`
+  display: flex;
+  flex-direction: row;
+  padding: 0 0 15px 0;
+  margin: 0 0 15px 0;
+  border: solid #d6d5d5;
+  border-width: 0 0 1px 0;
+`;
+
+const RoomIcon = styled.div`
+  border-radius: 50%;
+  background: gray;
+  width: 60px;
+  height: 60px;
+  margin-right: 20px;
+  flex-shrink: 0;
+`;
+
+const RoomInfoRow = styled.div`
+  display: flex;
+  flex-direction: row;
+  justify-content: space-between;
+  align-items: center;
+  width: 100%;
+  padding: 5px 0;
+`;
+
+const RoomInfoColumn = styled.div`
+  display: flex;
+  flex-direction: column;
+  justify-content: space-between;
+  flex: 1 0 auto;
+  max-width: calc(100% - 80px);
+`;
+
+const RoomName = styled.div`
+  font-size: 1.3em;
+  font-weight: bold;
+`;
+const LastChatTime = styled.div`
+  font-size: 0.8em;
+  color: gray;
+`;
+
+const LastChat = styled.div`
+  font-size: 1em;
+  color: gray;
+  overflow-wrap: break-word;
+`;
+
+export default function ChatRoomListItem({ room }) {
+  let lastChat = '';
+  if (room.chats[room.chats.length - 1].type === 'chat') {
```

Comment:
```suggestion
  if (room?.chats[room.chats.length - 1]?.type === 'chat') {
```

팁을 드리자면 객체의 속성을 여러 단계를 파고 들어가면( `a.b.c.d` ) Null Pointer 오류가 나서 스레드가 죽어버릴 가능성이 있어요. 그래서 [Optional Chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining)이라는 문법을 제공한답니다.

물론 이 코드가 동작하지 않는 것은 아니지만, 이런 기능이 있다면 잠재적 Null인 객체 혹은 배열이 다중으로 있는 상황에서는 유용하게 써보실 수 있을 것 같습니다 ㅎㅎ

Optional Chaining이라는 개념은 언어마다 다르지만, 안드로이드 진영에서 Java -> Kotlin으로 넘어가면서 개선되었던 Java의 NullPointerException의 고질적 문제점을 개선한것으로 유명하다고 하네요 ㅎㅎ